# @o3r/matching-configuration-name
Ensures that the configuration interface name matches the first parameter of `computeConfigurationName` used beside the configuration interface to expose its ID (as generated by the configuration module).

## How to use

```json
{
  "@o3r/o3r-widget-tags": "error"
}
```

## Valid code example

```typescript
import { computeItemIdentifier } from '@o3r/core';
import { Configuration } from '@o3r/core';

export interface MyFirstConfig extends Configuration {}

export const MY_CONFIG_ID = computeItemIdentifier('MyFirstConfig', '@example/config');
```

## Invalid code example

```typescript
import { computeItemIdentifier } from '@o3r/core';
import { Configuration } from '@o3r/core';

export interface MyConfig extends Configuration {}

export const MY_CONFIG_ID = computeItemIdentifier('InvalidName', '@example/config');
```
